### ISSUE TITLE
Update Makefile to check and warn if az cli if unavailable in local

### ISSUE THREAD

**Issue by nawazkh** [kubernetes-sigs/cluster-api-provider-azure#5291]
With https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5288 being added to CAPZ, tiltfile will depend on `az` commands to peer mgmt and workload cluster VNets. 
We should be updating `make tilt-up` command (probably as an additional dependency? )to check and warn the user `az` cli if not present in the local.

**Comment by brettearle** [comment]
I'm happy to jump on this

**Comment by willie-yao** [comment]
/assign @brettearle 

Hi! Just wondering if you were still planning to work on this?

**Comment by brettearle** [comment]
Yeah I'll jump on it, is there a discord/slack etc for questions or is here the best?

**Comment by willie-yao** [comment]
Sweet thank you! You can find us in the Kubernetes Slack: https://github.com/kubernetes-sigs/cluster-api-provider-azure?tab=readme-ov-file#getting-help

**Comment by brettearle** [comment]
@willie-yao Hey mate, I have a couple of questions.

Are we covering all 3 os's? 

Az has an install script for Linux. The script depends on openssl, python 3.8+, and libffi. What would you like to do when these are not available? I saw that the project supports Python 2.7, which will not work with this.

Do we want to reach across the network to get the install script or keep the script locally for the project?

**Comment by willie-yao** [comment]
Hi @brettearle!

Yes we should be covering all 3 os's

I think those are just a pre-requisite for CAPZ and your command shouldn't need to account for them.

If possible, it would be better to go across the network to get the install script!

If you'd like an example PR for something similar to this, check out #3630. @nawazkh Can you verify if the logic will largely be the same?

**Comment by nawazkh** [comment]
Hey @brettearle and @willie-yao ! Great depth analysis to this challenge here. 

The aim of this issue is to let the user know _"hey <user>, you do not have `azcli` installed in your local. Please install it before moving forward."_

So the way I see it, this issue can be broken down to two goals. 
1. Check if az is installed in user's local. 
2. Install az cli if not present. 

Not that I think of it with your comment @brettearle (https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/5291#issuecomment-2545240139), installation of azcli _could be unnecessary_. User might have their own preference on installing a cli tool, `brew`, `apt-get`, `macports`, `curl and move` etc. 
So, I think, we can **drop** the second goal of this issue of installing azcli and reduce the scope to first goal; "check and warn if az cli is not installed in local".

I will update this issues description as well.

**Comment by vishalanarase** [comment]
This issues seems to be pending long. I am happy to pick it up!
/assign
